packages/wmr/src/plugins/json-plugin.js

```diff
@@ -15,37 +13,25 @@ import { promises as fs } from 'fs';
  */
 export default function jsonPlugin({ root }) {
 	const IMPORT_PREFIX = 'json:';
-	const INTERNAL_PREFIX = '\0json:';
 
 	return {
 		name: 'json-plugin',
 		async resolveId(id, importer) {
-			if (!id.startsWith(IMPORT_PREFIX)) return;
-
-			id = id.slice(IMPORT_PREFIX.length);
+			if (id.startsWith(IMPORT_PREFIX)) {
+				id = id.slice(IMPORT_PREFIX.length);
+			}
+			if (!id.endsWith('.json')) return;
 
 			const resolved = await this.resolve(id, importer, { skipSelf: true });
-			return resolved && INTERNAL_PREFIX + resolved.id;
+			return resolved && resolved.id;
```

Comment:
A thought: what if we returned `${resolved.id}?json` here? (or `${resolved.id}?module`?)

Reply:
We currently add `?module` to any script file that's imported from a script. I feel like that's a more reliable approach as this allows us to differentiate urls that were fetched directly via `fetch()` calls vs urls that were loaded by an import statement.